pkg/trace: `TestInfoHandler` is too restrictive

`TestInfoHandler` is too restrictive. It's checking that the entire output of /info doesn't change, when it really wants to just be checking that the keys don't change. That means that if we add a new endpoint, we end up with errors like this
```
go test ./... -run=TestInfoHandler
--- FAIL: TestInfoHandler (0.00s)
    --- FAIL: TestInfoHandler/default (0.00s)
        info_test.go:243: 
                Error Trace:    /Users/katie.hockman/go/src/github.com/DataDog/datadog-agent/pkg/trace/api/info_test.go:243
                Error:          Not equal: 
                                expected: "{\n\t\"version\": \"0.99.0\",\n\t\"git_commit\": \"fab047e10\",\n\t\"endpoints\": [\n\t\t\"/v0.3/traces\",\n\t\t\"/v0.3/services\",\n\t\t\"/v0.4/traces\",\n\t\t\"/v0.4/services\",\n\t\t\"/v0.5/traces\",\n\t\t\"/v0.7/traces\",\n\t\t\"/profiling/v1/input\",\n\t\t\"/telemetry/proxy/\",\n\t\t\"/v0.6/stats\",\n\t\t\"/v0.1/pipeline_stats\",\n\t\t\"/appsec/proxy/\",\n\t\t\"/evp_proxy/v1/\",\n\t\t\"/debugger/v1/input\"\n\t\t\"/dogstatsd/v1/proxy\"\n\t],\n\t\"feature_flags\": [\n\t\t\"feature_flag\"\n\t],\n\t\"client_drop_p0s\": true,\n\t\"span_meta_structs\": true,\n\t\"long_running_spans\": true,\n\t\"config\": {\n\t\t\"default_env\": \"prod\",\n\t\t\"target_tps\": 11,\n\t\t\"max_eps\": 12,\n\t\t\"receiver_port\": 8111,\n\t\t\"receiver_socket\": \"/sock/path\",\n\t\t\"connection_limit\": 12,\n\t\t\"receiver_timeout\": 100,\n\t\t\"max_request_bytes\": 123,\n\t\t\"statsd_port\": 123,\n\t\t\"max_memory\": 1000000,\n\t\t\"max_cpu\": 12345,\n\t\t\"analyzed_spans_by_service\": {\n\t\t\t\"X\": {\n\t\t\t\t\"Y\": 2.4\n\t\t\t}\n\t\t},\n\t\t\"obfuscation\": {\n\t\t\t\"elastic_search\": true,\n\t\t\t\"mongo\": true,\n\t\t\t\"sql_exec_plan\": true,\n\t\t\t\"sql_exec_plan_normalize\": true,\n\t\t\t\"http\": {\n\t\t\t\t\"remove_query_string\": true,\n\t\t\t\t\"remove_path_digits\": true\n\t\t\t},\n\t\t\t\"remove_stack_traces\": false,\n\t\t\t\"redis\": true,\n\t\t\t\"memcached\": false\n\t\t}\n\t}\n}"
                                actual  : "{\n\t\"version\": \"0.99.0\",\n\t\"git_commit\": \"fab047e10\",\n\t\"endpoints\": [\n\t\t\"/v0.3/traces\",\n\t\t\"/v0.3/services\",\n\t\t\"/v0.4/traces\",\n\t\t\"/v0.4/services\",\n\t\t\"/v0.5/traces\",\n\t\t\"/v0.7/traces\",\n\t\t\"/profiling/v1/input\",\n\t\t\"/telemetry/proxy/\",\n\t\t\"/v0.6/stats\",\n\t\t\"/v0.1/pipeline_stats\",\n\t\t\"/appsec/proxy/\",\n\t\t\"/evp_proxy/v1/\",\n\t\t\"/debugger/v1/input\",\n\t\t\"/dogstatsd/v1/proxy\"\n\t],\n\t\"feature_flags\": [\n\t\t\"feature_flag\"\n\t],\n\t\"client_drop_p0s\": true,\n\t\"span_meta_structs\": true,\n\t\"long_running_spans\": true,\n\t\"config\": {\n\t\t\"default_env\": \"prod\",\n\t\t\"target_tps\": 11,\n\t\t\"max_eps\": 12,\n\t\t\"receiver_port\": 8111,\n\t\t\"receiver_socket\": \"/sock/path\",\n\t\t\"connection_limit\": 12,\n\t\t\"receiver_timeout\": 100,\n\t\t\"max_request_bytes\": 123,\n\t\t\"statsd_port\": 123,\n\t\t\"max_memory\": 1000000,\n\t\t\"max_cpu\": 12345,\n\t\t\"analyzed_spans_by_service\": {\n\t\t\t\"X\": {\n\t\t\t\t\"Y\": 2.4\n\t\t\t}\n\t\t},\n\t\t\"obfuscation\": {\n\t\t\t\"elastic_search\": true,\n\t\t\t\"mongo\": true,\n\t\t\t\"sql_exec_plan\": true,\n\t\t\t\"sql_exec_plan_normalize\": true,\n\t\t\t\"http\": {\n\t\t\t\t\"remove_query_string\": true,\n\t\t\t\t\"remove_path_digits\": true\n\t\t\t},\n\t\t\t\"remove_stack_traces\": false,\n\t\t\t\"redis\": true,\n\t\t\t\"memcached\": false\n\t\t}\n\t}\n}"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -16,3 +16,3 @@
                                                "/evp_proxy/v1/",
                                -               "/debugger/v1/input"
                                +               "/debugger/v1/input",
                                                "/dogstatsd/v1/proxy"
                Test:           TestInfoHandler/default
        info_test.go:245: Output of /info has changed. Changing the keys is not allowed because the client rely on them and is considered a breaking change:
            
            {
                "version": "0.99.0",
                "git_commit": "fab047e10",
                "endpoints": [
                        "/v0.3/traces",
                        "/v0.3/services",
                        "/v0.4/traces",
                        "/v0.4/services",
                        "/v0.5/traces",
                        "/v0.7/traces",
                        "/profiling/v1/input",
                        "/telemetry/proxy/",
                        "/v0.6/stats",
                        "/v0.1/pipeline_stats",
                        "/appsec/proxy/",
                        "/evp_proxy/v1/",
                        "/debugger/v1/input",
                        "/dogstatsd/v1/proxy"
                ],
                "feature_flags": [
                        "feature_flag"
                ],
                "client_drop_p0s": true,
                "span_meta_structs": true,
                "long_running_spans": true,
                "config": {
                        "default_env": "prod",
                        "target_tps": 11,
                        "max_eps": 12,
                        "receiver_port": 8111,
                        "receiver_socket": "/sock/path",
                        "connection_limit": 12,
                        "receiver_timeout": 100,
                        "max_request_bytes": 123,
                        "statsd_port": 123,
                        "max_memory": 1000000,
                        "max_cpu": 12345,
                        "analyzed_spans_by_service": {
                                "X": {
                                        "Y": 2.4
                                }
                        },
                        "obfuscation": {
                                "elastic_search": true,
                                "mongo": true,
                                "sql_exec_plan": true,
                                "sql_exec_plan_normalize": true,
                                "http": {
                                        "remove_query_string": true,
                                        "remove_path_digits": true
                                },
                                "remove_stack_traces": false,
                                "redis": true,
                                "memcached": false
                        }
                }
            }
    --- FAIL: TestInfoHandler/debug (0.00s)
        info_test.go:243: 
                Error Trace:    /Users/katie.hockman/go/src/github.com/DataDog/datadog-agent/pkg/trace/api/info_test.go:243
                Error:          Not equal: 
                                expected: "{\n\t\"version\": \"0.99.0\",\n\t\"git_commit\": \"fab047e10\",\n\t\"endpoints\": [\n\t\t\"/v0.3/traces\",\n\t\t\"/v0.3/services\",\n\t\t\"/v0.4/traces\",\n\t\t\"/v0.4/services\",\n\t\t\"/v0.5/traces\",\n\t\t\"/v0.7/traces\",\n\t\t\"/profiling/v1/input\",\n\t\t\"/telemetry/proxy/\",\n\t\t\"/v0.6/stats\",\n\t\t\"/v0.1/pipeline_stats\",\n\t\t\"/appsec/proxy/\",\n\t\t\"/evp_proxy/v1/\",\n\t\t\"/debugger/v1/input\"\n\t],\n\t\"feature_flags\": [\n\t\t\"feature_flag\"\n\t],\n\t\"client_drop_p0s\": true,\n\t\"span_meta_structs\": true,\n\t\"long_running_spans\": true,\n\t\"config\": {\n\t\t\"default_env\": \"prod\",\n\t\t\"target_tps\": 11,\n\t\t\"max_eps\": 12,\n\t\t\"receiver_port\": 8111,\n\t\t\"receiver_socket\": \"/sock/path\",\n\t\t\"connection_limit\": 12,\n\t\t\"receiver_timeout\": 100,\n\t\t\"max_request_bytes\": 123,\n\t\t\"statsd_port\": 123,\n\t\t\"max_memory\": 1000000,\n\t\t\"max_cpu\": 12345,\n\t\t\"analyzed_spans_by_service\": {\n\t\t\t\"X\": {\n\t\t\t\t\"Y\": 2.4\n\t\t\t}\n\t\t},\n\t\t\"obfuscation\": {\n\t\t\t\"elastic_search\": true,\n\t\t\t\"mongo\": true,\n\t\t\t\"sql_exec_plan\": true,\n\t\t\t\"sql_exec_plan_normalize\": true,\n\t\t\t\"http\": {\n\t\t\t\t\"remove_query_string\": true,\n\t\t\t\t\"remove_path_digits\": true\n\t\t\t},\n\t\t\t\"remove_stack_traces\": false,\n\t\t\t\"redis\": true,\n\t\t\t\"memcached\": false\n\t\t}\n\t}\n}"
                                actual  : "{\n\t\"version\": \"0.99.0\",\n\t\"git_commit\": \"fab047e10\",\n\t\"endpoints\": [\n\t\t\"/v0.3/traces\",\n\t\t\"/v0.3/services\",\n\t\t\"/v0.4/traces\",\n\t\t\"/v0.4/services\",\n\t\t\"/v0.5/traces\",\n\t\t\"/v0.7/traces\",\n\t\t\"/profiling/v1/input\",\n\t\t\"/telemetry/proxy/\",\n\t\t\"/v0.6/stats\",\n\t\t\"/v0.1/pipeline_stats\",\n\t\t\"/appsec/proxy/\",\n\t\t\"/evp_proxy/v1/\",\n\t\t\"/debugger/v1/input\",\n\t\t\"/dogstatsd/v1/proxy\"\n\t],\n\t\"feature_flags\": [\n\t\t\"feature_flag\"\n\t],\n\t\"client_drop_p0s\": true,\n\t\"span_meta_structs\": true,\n\t\"long_running_spans\": true,\n\t\"config\": {\n\t\t\"default_env\": \"prod\",\n\t\t\"target_tps\": 11,\n\t\t\"max_eps\": 12,\n\t\t\"receiver_port\": 8111,\n\t\t\"receiver_socket\": \"/sock/path\",\n\t\t\"connection_limit\": 12,\n\t\t\"receiver_timeout\": 100,\n\t\t\"max_request_bytes\": 123,\n\t\t\"statsd_port\": 123,\n\t\t\"max_memory\": 1000000,\n\t\t\"max_cpu\": 12345,\n\t\t\"analyzed_spans_by_service\": {\n\t\t\t\"X\": {\n\t\t\t\t\"Y\": 2.4\n\t\t\t}\n\t\t},\n\t\t\"obfuscation\": {\n\t\t\t\"elastic_search\": true,\n\t\t\t\"mongo\": true,\n\t\t\t\"sql_exec_plan\": true,\n\t\t\t\"sql_exec_plan_normalize\": true,\n\t\t\t\"http\": {\n\t\t\t\t\"remove_query_string\": true,\n\t\t\t\t\"remove_path_digits\": true\n\t\t\t},\n\t\t\t\"remove_stack_traces\": false,\n\t\t\t\"redis\": true,\n\t\t\t\"memcached\": false\n\t\t}\n\t}\n}"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -16,3 +16,4 @@
                                                "/evp_proxy/v1/",
                                -               "/debugger/v1/input"
                                +               "/debugger/v1/input",
                                +               "/dogstatsd/v1/proxy"
                                        ],
                Test:           TestInfoHandler/debug
        info_test.go:245: Output of /info has changed. Changing the keys is not allowed because the client rely on them and is considered a breaking change:
            
            {
                "version": "0.99.0",
                "git_commit": "fab047e10",
                "endpoints": [
                        "/v0.3/traces",
                        "/v0.3/services",
                        "/v0.4/traces",
                        "/v0.4/services",
                        "/v0.5/traces",
                        "/v0.7/traces",
                        "/profiling/v1/input",
                        "/telemetry/proxy/",
                        "/v0.6/stats",
                        "/v0.1/pipeline_stats",
                        "/appsec/proxy/",
                        "/evp_proxy/v1/",
                        "/debugger/v1/input",
                        "/dogstatsd/v1/proxy"
                ],
                "feature_flags": [
                        "feature_flag"
                ],
                "client_drop_p0s": true,
                "span_meta_structs": true,
                "long_running_spans": true,
                "config": {
                        "default_env": "prod",
                        "target_tps": 11,
                        "max_eps": 12,
                        "receiver_port": 8111,
                        "receiver_socket": "/sock/path",
                        "connection_limit": 12,
                        "receiver_timeout": 100,
                        "max_request_bytes": 123,
                        "statsd_port": 123,
                        "max_memory": 1000000,
                        "max_cpu": 12345,
                        "analyzed_spans_by_service": {
                                "X": {
                                        "Y": 2.4
                                }
                        },
                        "obfuscation": {
                                "elastic_search": true,
                                "mongo": true,
                                "sql_exec_plan": true,
                                "sql_exec_plan_normalize": true,
                                "http": {
                                        "remove_query_string": true,
                                        "remove_path_digits": true
                                },
                                "remove_stack_traces": false,
                                "redis": true,
                                "memcached": false
                        }
                }
            }
FAIL
FAIL    github.com/DataDog/datadog-agent/pkg/trace/api  0.343s
ok      github.com/DataDog/datadog-agent/pkg/trace/api/apiutil  (cached) [no tests to run]
FAIL
```

--- a/pkg/trace/api/info_test.go
+++ b/pkg/trace/api/info_test.go
@@ -52,7 +52,7 @@ func ensureKeys(expect, result map[string]interface{}, prefix string) error {
 			}
 		}
 	}
-	for k, _ := range result {
+	for k := range result {
 		_, ok := expect[k]
 		if !ok {
 			path := k

--- a/pkg/trace/api/info_test.go
+++ b/pkg/trace/api/info_test.go
@@ -66,7 +66,6 @@ func ensureKeys(expect, result map[string]interface{}, prefix string) error {
 // * In case a field name gets modified, the `json:""` struct field tag
 // should be used to ensure the old key is marshalled for this endpoint.
 func TestInfoHandler(t *testing.T) {
-	t.Skip("https://github.com/DataDog/datadog-agent/issues/13569")
 	u, err := url.Parse("http://localhost:8888/proxy")
 	if err != nil {
 		log.Fatal(err)

--- a/pkg/trace/api/info_test.go
+++ b/pkg/trace/api/info_test.go
@@ -6,6 +6,9 @@
 package api
 
 import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"log"
 	"net/http/httptest"
 	"net/url"
@@ -17,6 +20,43 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/trace/config"
 	"github.com/DataDog/datadog-agent/pkg/trace/testutil"
 )
+
+func ensureKeys(expect, result map[string]interface{}, prefix string) error {
+	for k, ev := range expect {
+		rv, ok := result[k]
+		if !ok {
+			if prefix != "" {
+				k = prefix + "." + k
+			}
+			return fmt.Errorf("Expected key %s, but it is not present in the output.\n", k)
+		}
+
+		if em, ok := ev.(map[string]interface{}); ok {
+			rm, ok := rv.(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("Expected key %s to be a map, but it is '%#v'.\n", k, rv)
+			}
+			if prefix != "" {
+				prefix = prefix + "." + k
+			} else {
+				prefix = k
+			}
+			if err := ensureKeys(em, rm, prefix); err != nil {
+				return err
+			}
+		}
+	}
+	for k, _ := range result {
+		_, ok := expect[k]
+		if !ok {
+			if prefix != "" {
+				k = prefix + "." + k
+				return fmt.Errorf("Found key %s, but it is not expected in the output. If you've added a new key to the /info endpoint, please add it to the tests.\n", k)
+			}
+		}
+	}
+	return nil
+}
 
 // TestInfoHandler ensures that the keys returned by the /info handler do not
 // change from one release to another to ensure consistency. Tracing clients
@@ -109,146 +149,56 @@ func TestInfoHandler(t *testing.T) {
 		},
 	}
 
-	var testCases = []struct {
-		name     string
-		expected string
-	}{
-		{
-			name: "default",
-			expected: `{
-	"version": "0.99.0",
-	"git_commit": "fab047e10",
-	"endpoints": [
-		"/v0.3/traces",
-		"/v0.3/services",
-		"/v0.4/traces",
-		"/v0.4/services",
-		"/v0.5/traces",
-		"/v0.7/traces",
-		"/profiling/v1/input",
-		"/telemetry/proxy/",
-		"/v0.6/stats",
-		"/v0.1/pipeline_stats",
-		"/evp_proxy/v1/",
-		"/evp_proxy/v2/",
-		"/debugger/v1/input"
-		"/dogstatsd/v1/proxy"
-	],
-	"feature_flags": [
-		"feature_flag"
-	],
-	"client_drop_p0s": true,
-	"span_meta_structs": true,
-	"long_running_spans": true,
-	"config": {
-		"default_env": "prod",
-		"target_tps": 11,
-		"max_eps": 12,
-		"receiver_port": 8111,
-		"receiver_socket": "/sock/path",
-		"connection_limit": 12,
-		"receiver_timeout": 100,
-		"max_request_bytes": 123,
-		"statsd_port": 123,
-		"max_memory": 1000000,
-		"max_cpu": 12345,
-		"analyzed_spans_by_service": {
-			"X": {
-				"Y": 2.4
-			}
-		},
-		"obfuscation": {
-			"elastic_search": true,
-			"mongo": true,
-			"sql_exec_plan": true,
-			"sql_exec_plan_normalize": true,
-			"http": {
-				"remove_query_string": true,
-				"remove_path_digits": true
+	expectedKeys := map[string]interface{}{
+		"version":            nil,
+		"git_commit":         nil,
+		"endpoints":          nil,
+		"feature_flags":      nil,
+		"client_drop_p0s":    nil,
+		"span_meta_structs":  nil,
+		"long_running_spans": nil,
+		"config": map[string]interface{}{
+			"default_env":               nil,
+			"target_tps":                nil,
+			"max_eps":                   nil,
+			"receiver_port":             nil,
+			"receiver_socket":           nil,
+			"connection_limit":          nil,
+			"receiver_timeout":          nil,
+			"max_request_bytes":         nil,
+			"statsd_port":               nil,
+			"max_memory":                nil,
+			"max_cpu":                   nil,
+			"analyzed_spans_by_service": nil,
+			"obfuscation": map[string]interface{}{
+				"elastic_search":          nil,
+				"mongo":                   nil,
+				"sql_exec_plan":           nil,
+				"sql_exec_plan_normalize": nil,
+				"http": map[string]interface{}{
+					"remove_query_string": nil,
+					"remove_path_digits":  nil,
+				},
+				"remove_stack_traces": nil,
+				"redis":               nil,
+				"memcached":           nil,
 			},
-			"remove_stack_traces": false,
-			"redis": true,
-			"memcached": false
-		}
-	}
-}`,
-		},
-		{
-			name: "debug",
-			expected: `{
-	"version": "0.99.0",
-	"git_commit": "fab047e10",
-	"endpoints": [
-		"/v0.3/traces",
-		"/v0.3/services",
-		"/v0.4/traces",
-		"/v0.4/services",
-		"/v0.5/traces",
-		"/v0.7/traces",
-		"/profiling/v1/input",
-		"/telemetry/proxy/",
-		"/v0.6/stats",
-		"/v0.1/pipeline_stats",
-		"/evp_proxy/v1/",
-		"/evp_proxy/v2/",
-		"/evp_proxy/v3/",
-		"/debugger/v1/input"
-		"/dogstatsd/v1/proxy"
-	],
-	"feature_flags": [
-		"feature_flag"
-	],
-	"client_drop_p0s": true,
-	"span_meta_structs": true,
-	"long_running_spans": true,
-	"config": {
-		"default_env": "prod",
-		"target_tps": 11,
-		"max_eps": 12,
-		"receiver_port": 8111,
-		"receiver_socket": "/sock/path",
-		"connection_limit": 12,
-		"receiver_timeout": 100,
-		"max_request_bytes": 123,
-		"statsd_port": 123,
-		"max_memory": 1000000,
-		"max_cpu": 12345,
-		"analyzed_spans_by_service": {
-			"X": {
-				"Y": 2.4
-			}
-		},
-		"obfuscation": {
-			"elastic_search": true,
-			"mongo": true,
-			"sql_exec_plan": true,
-			"sql_exec_plan_normalize": true,
-			"http": {
-				"remove_query_string": true,
-				"remove_path_digits": true
-			},
-			"remove_stack_traces": false,
-			"redis": true,
-			"memcached": false
-		}
-	}
-}`,
 		},
 	}
-	for _, tt := range testCases {
-		t.Run(tt.name, func(t *testing.T) {
-			rcv := newTestReceiverFromConfig(conf)
-			defer testutil.WithFeatures("feature_flag")()
-			_, h := rcv.makeInfoHandler()
-			rec := httptest.NewRecorder()
-			req := httptest.NewRequest("GET", "/info", nil)
-			h.ServeHTTP(rec, req)
-			assert.Equal(t, tt.expected, rec.Body.String())
-			if rec.Body.String() != tt.expected {
-				t.Fatalf("Output of /info has changed. Changing the keys "+
-					"is not allowed because the client rely on them and "+
-					"is considered a breaking change:\n\n%v", rec.Body.String())
-			}
-		})
+
+	rcv := newTestReceiverFromConfig(conf)
+	defer testutil.WithFeatures("feature_flag")()
+	_, h := rcv.makeInfoHandler()
+	rec := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", "/info", nil)
+	h.ServeHTTP(rec, req)
+	var m map[string]interface{}
+	b, err := ioutil.ReadAll(rec.Body)
+	if !assert.NoError(t, err) {
+		return
 	}
+	if !assert.NoError(t, json.Unmarshal(b, &m)) {
+		return
+	}
+	assert.NoError(t, ensureKeys(expectedKeys, m, ""))
 }

--- a/pkg/trace/api/info_test.go
+++ b/pkg/trace/api/info_test.go
@@ -21,14 +21,21 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/trace/testutil"
 )
 
+// ensureKeys takes 2 maps, expect and result, and ensures that the set of keys in expect and
+// result match. For each key (k) in expect, if expect[k] is of type map[string]interface{}, then
+// ensureKeys recurses on expect[k], result[k], prefix + "." + k.
+// 
+// This should ensure that whatever keys and maps are defined in expect are exactly mirrored in
+// result, but without checking for specific values in result.
 func ensureKeys(expect, result map[string]interface{}, prefix string) error {
 	for k, ev := range expect {
 		rv, ok := result[k]
 		if !ok {
+			path := k
 			if prefix != "" {
-				k = prefix + "." + k
+				path = prefix + "." + k
 			}
-			return fmt.Errorf("Expected key %s, but it is not present in the output.\n", k)
+			return fmt.Errorf("Expected key %s, but it is not present in the output.\n", path)
 		}
 
 		if em, ok := ev.(map[string]interface{}); ok {
@@ -49,10 +56,11 @@ func ensureKeys(expect, result map[string]interface{}, prefix string) error {
 	for k, _ := range result {
 		_, ok := expect[k]
 		if !ok {
+			path := k
 			if prefix != "" {
-				k = prefix + "." + k
-				return fmt.Errorf("Found key %s, but it is not expected in the output. If you've added a new key to the /info endpoint, please add it to the tests.\n", k)
+				path = prefix + "." + k
 			}
+			return fmt.Errorf("Found key %s, but it is not expected in the output. If you've added a new key to the /info endpoint, please add it to the tests.\n", path)
 		}
 	}
 	return nil

--- a/pkg/trace/api/info_test.go
+++ b/pkg/trace/api/info_test.go
@@ -8,7 +8,6 @@ package api
 import (
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"log"
 	"net/http/httptest"
 	"net/url"
@@ -337,11 +336,7 @@ func TestInfoHandler(t *testing.T) {
 	req := httptest.NewRequest("GET", "/info", nil)
 	h.ServeHTTP(rec, req)
 	var m map[string]interface{}
-	b, err := ioutil.ReadAll(rec.Body)
-	if !assert.NoError(t, err) {
-		return
-	}
-	if !assert.NoError(t, json.Unmarshal(b, &m)) {
+	if !assert.NoError(t, json.NewDecoder(rec.Body).Decode(&m)) {
 		return
 	}
 	assert.NoError(t, ensureKeys(expectedKeys, m, ""))

--- a/pkg/trace/api/info_test.go
+++ b/pkg/trace/api/info_test.go
@@ -84,8 +84,8 @@ func TestEnsureKeys(t *testing.T) {
 		},
 		{
 			expect: map[string]interface{}{
-				"one": nil,
-				"two": nil,
+				"one":   nil,
+				"two":   nil,
 				"three": nil,
 			},
 			result: map[string]interface{}{
@@ -100,8 +100,8 @@ func TestEnsureKeys(t *testing.T) {
 				"two": nil,
 			},
 			result: map[string]interface{}{
-				"one": 1,
-				"two": "two",
+				"one":   1,
+				"two":   "two",
 				"three": 3,
 			},
 			err: true,
@@ -110,7 +110,7 @@ func TestEnsureKeys(t *testing.T) {
 			expect: map[string]interface{}{
 				"one": nil,
 				"two": nil,
-				"sub": map[string]interface{} {
+				"sub": map[string]interface{}{
 					"subone": nil,
 					"subtwo": nil,
 				},
@@ -118,7 +118,7 @@ func TestEnsureKeys(t *testing.T) {
 			result: map[string]interface{}{
 				"one": 1,
 				"two": "two",
-				"sub": map[string]interface{} {
+				"sub": map[string]interface{}{
 					"subone": 1,
 					"subtwo": 2,
 				},
@@ -128,18 +128,18 @@ func TestEnsureKeys(t *testing.T) {
 			expect: map[string]interface{}{
 				"one": nil,
 				"two": nil,
-				"sub": map[string]interface{} {
+				"sub": map[string]interface{}{
 					"subone": nil,
 					"subtwo": nil,
 				},
 			},
 			result: map[string]interface{}{
 				"one": 1,
-				"two": map[string]interface{} { // Map values not described in expect are NOT checked, so this is OK.
+				"two": map[string]interface{}{ // Map values not described in expect are NOT checked, so this is OK.
 					"subone": 1,
 					"subtwo": 2,
 				},
-				"sub": map[string]interface{} {
+				"sub": map[string]interface{}{
 					"subone": 1,
 					"subtwo": 2,
 				},
@@ -149,19 +149,19 @@ func TestEnsureKeys(t *testing.T) {
 			expect: map[string]interface{}{
 				"one": nil,
 				"two": nil,
-				"sub": map[string]interface{} {
-					"subone": nil,
-					"subtwo": nil,
+				"sub": map[string]interface{}{
+					"subone":   nil,
+					"subtwo":   nil,
 					"subthree": nil,
 				},
 			},
 			result: map[string]interface{}{
 				"one": 1,
-				"two": map[string]interface{} { // Map values not described in expect are NOT checked, so this is OK.
+				"two": map[string]interface{}{ // Map values not described in expect are NOT checked, so this is OK.
 					"subone": 1,
 					"subtwo": 2,
 				},
-				"sub": map[string]interface{} {
+				"sub": map[string]interface{}{
 					"subone": 1,
 					"subtwo": 2,
 				},
@@ -172,20 +172,20 @@ func TestEnsureKeys(t *testing.T) {
 			expect: map[string]interface{}{
 				"one": nil,
 				"two": nil,
-				"sub": map[string]interface{} {
+				"sub": map[string]interface{}{
 					"subone": nil,
 					"subtwo": nil,
 				},
 			},
 			result: map[string]interface{}{
 				"one": 1,
-				"two": map[string]interface{} { // Map values not described in expect are NOT checked, so this is OK.
+				"two": map[string]interface{}{ // Map values not described in expect are NOT checked, so this is OK.
 					"subone": 1,
 					"subtwo": 2,
 				},
-				"sub": map[string]interface{} {
-					"subone": 1,
-					"subtwo": 2,
+				"sub": map[string]interface{}{
+					"subone":   1,
+					"subtwo":   2,
 					"subthree": 3,
 				},
 			},

--- a/pkg/trace/api/info_test.go
+++ b/pkg/trace/api/info_test.go
@@ -24,7 +24,7 @@ import (
 // ensureKeys takes 2 maps, expect and result, and ensures that the set of keys in expect and
 // result match. For each key (k) in expect, if expect[k] is of type map[string]interface{}, then
 // ensureKeys recurses on expect[k], result[k], prefix + "." + k.
-// 
+//
 // This should ensure that whatever keys and maps are defined in expect are exactly mirrored in
 // result, but without checking for specific values in result.
 func ensureKeys(expect, result map[string]interface{}, prefix string) error {
@@ -64,6 +64,143 @@ func ensureKeys(expect, result map[string]interface{}, prefix string) error {
 		}
 	}
 	return nil
+}
+
+func TestEnsureKeys(t *testing.T) {
+	for _, tt := range []struct {
+		expect map[string]interface{}
+		result map[string]interface{}
+		err    bool
+	}{
+		{
+			expect: map[string]interface{}{
+				"one": nil,
+				"two": nil,
+			},
+			result: map[string]interface{}{
+				"one": 1,
+				"two": "two",
+			},
+		},
+		{
+			expect: map[string]interface{}{
+				"one": nil,
+				"two": nil,
+				"three": nil,
+			},
+			result: map[string]interface{}{
+				"one": 1,
+				"two": "two",
+			},
+			err: true,
+		},
+		{
+			expect: map[string]interface{}{
+				"one": nil,
+				"two": nil,
+			},
+			result: map[string]interface{}{
+				"one": 1,
+				"two": "two",
+				"three": 3,
+			},
+			err: true,
+		},
+		{
+			expect: map[string]interface{}{
+				"one": nil,
+				"two": nil,
+				"sub": map[string]interface{} {
+					"subone": nil,
+					"subtwo": nil,
+				},
+			},
+			result: map[string]interface{}{
+				"one": 1,
+				"two": "two",
+				"sub": map[string]interface{} {
+					"subone": 1,
+					"subtwo": 2,
+				},
+			},
+		},
+		{
+			expect: map[string]interface{}{
+				"one": nil,
+				"two": nil,
+				"sub": map[string]interface{} {
+					"subone": nil,
+					"subtwo": nil,
+				},
+			},
+			result: map[string]interface{}{
+				"one": 1,
+				"two": map[string]interface{} { // Map values not described in expect are NOT checked, so this is OK.
+					"subone": 1,
+					"subtwo": 2,
+				},
+				"sub": map[string]interface{} {
+					"subone": 1,
+					"subtwo": 2,
+				},
+			},
+		},
+		{
+			expect: map[string]interface{}{
+				"one": nil,
+				"two": nil,
+				"sub": map[string]interface{} {
+					"subone": nil,
+					"subtwo": nil,
+					"subthree": nil,
+				},
+			},
+			result: map[string]interface{}{
+				"one": 1,
+				"two": map[string]interface{} { // Map values not described in expect are NOT checked, so this is OK.
+					"subone": 1,
+					"subtwo": 2,
+				},
+				"sub": map[string]interface{} {
+					"subone": 1,
+					"subtwo": 2,
+				},
+			},
+			err: true,
+		},
+		{
+			expect: map[string]interface{}{
+				"one": nil,
+				"two": nil,
+				"sub": map[string]interface{} {
+					"subone": nil,
+					"subtwo": nil,
+				},
+			},
+			result: map[string]interface{}{
+				"one": 1,
+				"two": map[string]interface{} { // Map values not described in expect are NOT checked, so this is OK.
+					"subone": 1,
+					"subtwo": 2,
+				},
+				"sub": map[string]interface{} {
+					"subone": 1,
+					"subtwo": 2,
+					"subthree": 3,
+				},
+			},
+			err: true,
+		},
+	} {
+		t.Run("", func(t *testing.T) {
+			err := ensureKeys(tt.expect, tt.result, "")
+			if tt.err {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
 }
 
 // TestInfoHandler ensures that the keys returned by the /info handler do not

